boot/bootutil/src/encrypted.c

```diff
@@ -415,7 +387,7 @@ static int fake_rng(void *p_rng, unsigned char *output, size_t len)
  * @param enckey An AES-128 or AES-256 key sized buffer to store to plain key.
  */
 int
-boot_enc_decrypt(const uint8_t *buf, uint8_t *enckey)
+boot_decrypt_key(const uint8_t *buf, uint8_t *enckey)
```

Comment:
Why not moving these functions out of `encrypted.c` for better code organization? In this case you would have 2 source files implementing these functions, i.e. `encrypted_psa.c` and the new one, without a huge `#if !defined(CONFIG_BOOT_ED25519_PSA)` in `encrypted.c`...

Reply:
I do not want to do that within this PR, that is something I would like to do, but that has to be done upstream.
Because we are trying to do things with [nrf noups] till we are able to push the changes upstream, I do not want to divert from the upstream code too much.